File: Data Structures & Algorithms/products-of-array-discluding-self/submission-4.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        left = [1]*len(nums)
        right=[1]*len(nums)
        output=[1]*len(nums)

        for i in range(len(nums)):
            if i==0:
                left[i]=1
            else:
                left[i]=left[i-1]*nums[i-1]
       
        for i in range(len(nums)-1,-1,-1):
            if i==len(nums)-1:
                right[i]=1
            else:
                right[i]=nums[i+1]*right[i+1]
        
        for i in range(len(left)):
            output[i]=left[i]*right[i]
        return output
        

File: Data Structures & Algorithms/is-palindrome/submission-4.java
class Solution {
    public boolean isPalindrome(String s) {
        
        int l = 0;
        int r = s.length()-1;

        while(l<r){
            char left = s.charAt(l);
            char right = s.charAt(r);
            if(!Character.isLetterOrDigit(left)){
                l+=1;
            }
            else if(!Character.isLetterOrDigit(right)){
                r-=1;
            }
            else{
                if(Character.toLowerCase(left)!=Character.toLowerCase(right)){
                    return false;
                }
                l+=1;
                r-=1;
            }
        }
        return true;
        
    }
}
